xmppserver/src/main/java/org/jivesoftware/openfire/roster/RosterManager.java

```diff
@@ -361,16 +361,13 @@ else if ("sharedRoster.displayName".equals(keyChanged)) {
             // Iterate on all the affected users and update their rosters
             for (JID updatedUser : users) {
                 // Get the roster to update.
-                Roster roster;
+                Roster roster = null;
                 if (server.isLocal(updatedUser)) {
-                    try {
-                        roster = getRoster(updatedUser.getNode());
-
-                        // Update the roster with the new group display name
-                        roster.shareGroupRenamed(users);
-                    } catch (UserNotFoundException e) {
-                        Log.debug( "Unexpected exception while applying group modification for user '{}' .", updatedUser.getNode(), e );
-                    }
+                    roster = rosterCache.get(updatedUser.getNode());
```

Comment:
This only considers a roster that's already loaded in the cache, but not ones that are only in the database. Is this not a concern because we're processing a 'shared roster' change here (which perhaps does not affect data that's stored on the roster entry of individual users in the database)?

Reply:
Maybe shared roster entries are constructed on demand until manual roster entries. I would need to dive much deeper into the code to confirm that. As this was the previous code, we have to consider that it has some merit.